Sources/SwiftSyntax/Identifier.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// An abstraction for sanitized values on a token.
+public struct Identifier: Equatable, Hashable, Sendable {
+    /// The sanitized `text` of a token.
+    public var name: String {
+        String(syntaxText: rawIdentifier.name)
+    }
+    
+    @_spi(RawSyntax)
+    public let rawIdentifier: RawIdentifier
+    
+    let arena: RetainedSyntaxArena
+    
+    public init?(_ token: TokenSyntax) {
+        guard case .identifier(let text) = token.tokenKind else {
+            return nil
+        }
+        
+        var rawText = text.contains("`") ? text.trimmingCharacters(in: "`") : Substring(text)
+        
+        let syntaxArena = SyntaxArena()
+        
+        let name = rawText.withUTF8 {
+            syntaxArena.intern(
+                SyntaxText(buffer: SyntaxArenaAllocatedBufferPointer<UInt8>($0))
+            )
+        }
```

Comment:
I don’t think we need to create a new arena and intern the string at all. If you pick up my suggestion of doing the trimming in `RawIdentifier`, you should be able to
- Get the `RawSyntaxTokenView.rawText`
- If the text has a leading and trailing backtick, slice off the first and last byte of the `SyntaxText` using a subscript
- Constructing a `SyntaxText` from the slice again using `SyntaxText.init(rebasing:)`

What’s conceptually happening then, is that the whole text including backticks is allocated memory in the `SyntaxArena` and `RawIdentifier` just references the slice of that text without the backticks without doing any more memory allocations.

---

Also, I think we don’t want to trim an arbitrary number of backticks at the front and back, but a single backtick at the front and a single backtick at the back if both exist.

Reply:
I think these comments have now been resolved, let me know what you think.